### PR TITLE
LPD-29830 Adapt ThreadLocal to use the id of the object entry so that validations will run properly in a batch call

### PR DESCRIPTION
**Pull request by devpedrotavares**
Related ticket: [LPD-29830](https://liferay.atlassian.net/browse/LPD-29830)

[LPD-29830]: https://liferay.atlassian.net/browse/LPD-29830?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java [hunk 1]
```diff
@@ -455,6 +455,61 @@ public void tearDown() throws Exception {
...
+				"listTypeEntryKeyRequired", "listTypeEntryKey1"
+			).build());
+
+		_assertCount(1);
+
+		for (String invalidEmailAddress :
+				Arrays.asList(
+					RandomTestUtil.randomString(),
+					RandomTestUtil.randomString(),
+					RandomTestUtil.randomString())) {
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java:480]: @devpedrotavares do we need 3 random strings or we can just use 2?

**Review by MarinhoFeliphe — APPROVED** [review]
LGTM!! :tada:

Good job, @devpedrotavares!

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:object

**Comment by marcelabc** [comment]
ci:report:29023470

**Comment by marcelabc** [comment]
ci:report:29008455

**Comment by marcelabc** [comment]
From `ci:test:object`, There are 2 unique failures that need to be investigated:

- LocalFile.ObjectUpgrade#ViewObjectRelationshipAfterUpgrade7413U33
- LocalFile.CreateObject#CanPublishObjectWithPublishPermission

I'm running CI again just to compare these failures.

**Comment by marcelabc** [comment]
ci:test:object

**Comment by marcelabc** [comment]
ci:report:30232700

**Comment by marcelabc** [comment]
The failures are not related because it has passed in the latest run.

**Comment by marcelabc** [comment]
ci:forward:force

**Pull request closed without merging**